Update build.md (#1221)
3scale/APIcast: doc/build.md

# Build process and components

> APIcast is [OpenResty](https://openresty.org/en/) application. It consists of two parts: nginx configuration and lua files.

Reviewer: ```suggestion
APIcast is an application based on [OpenResty](https://openresty.org/en/). APIcast consists of two parts: NGINX configuration and Lua files.
```

## Dependencies

> APIcast uses LuaRocks to install dependencies. LuaRocks have to be properly configured,

Reviewer: ```suggestion
APIcast uses LuaRocks, the package manager for Lua modules, to install dependencies. With a correct configuration, LuaRocks 
```

> so it installs dependencies into the correct path where OpenResty can see them. In the Docker image, rocks are installed into the application folder.

Reviewer: ```suggestion
so it installs dependencies into the correct path where OpenResty can see them. For Docker images, LuaRocks is installed into the application folder.
```

> Then this folder is added to the load path by `luarocks path`. 

Reviewer: What does `this folder` refer to:
* the application folder of Docker Images
or
* the correct path where OpenResty sees the dependencies

Author: It is "the application folder of Docker Images " from what I understand and interpret.

Reviewer: In this case, we need to restructure the content as the following:

```
APIcast uses LuaRocks, the package manager for Lua modules, to install dependencies. With a correct configuration, LuaRocks installs dependencies into the correct path where OpenResty can see them. 

For Docker images, LuaRocks is installed into the application folder. Then, `luarocks path` adds the application folder to the load path.
```

Does this make sense?

Author: Looks ok to me. Made the changes.

You'll need to have source-to-image installed and available on your system.

The builder image used is [s2i-openresty](https://github.com/3scale/s2i-openresty).

> It is not a very light builder image as it builds on heavy openshift base images.

Reviewer: ```suggestion
It builds on heavy OpenShift base images.
```
